src/HealthGPS.Input/data_source.cpp

```diff
@@ -36,73 +20,50 @@ std::filesystem::path get_zip_file_path(const std::string &source) {
         "Data source must be a directory, a zip file or a URL pointing to a zip file");
 }
 
-std::filesystem::path get_data_directory_with_validation(const std::string &source,
-                                                         const std::string &file_hash) {
-    // If the cache folder already exists, then we don't need to download or extract anything
-    auto cache_path = hgps::input::get_zip_cache_directory(file_hash);
-    if (std::filesystem::is_directory(cache_path)) {
-        return cache_path;
-    }
-
-    const auto zip_file_path = get_zip_file_path(source);
-
-    // Validate file with checksum
-    const auto computed_hash = hgps::compute_sha256_for_file(zip_file_path);
-    if (computed_hash != file_hash) {
-        throw std::runtime_error(
-            fmt::format("Checksum validation failed for {} (actual: {}, expected: {})", source,
-                        computed_hash, file_hash));
+// If source is a relative path, rebase it on root_path, else just return source
+std::string try_rebase_path(std::string source, const std::filesystem::path &root_path) {
+    // If source is a URL, leave it alone
+    if (source.starts_with("http://") || source.starts_with("https://")) {
+        return source;
     }
 
-    // Extract files
-    hgps::input::extract_zip_file(zip_file_path, cache_path);
-
-    return cache_path;
-}
-
-std::filesystem::path get_data_directory_without_validation(const std::string &source) {
-    const auto zip_file_path = get_zip_file_path(source);
-    const auto file_hash = hgps::compute_sha256_for_file(zip_file_path);
-
-    // If the cache folder already exists, then we don't need to extract anything
-    auto cache_path = hgps::input::get_zip_cache_directory(file_hash);
-    if (std::filesystem::is_directory(cache_path)) {
-        return cache_path;
+    const std::filesystem::path path = source;
+    if (path.is_absolute()) {
+        return source;
     }
 
-    // Extract files
-    hgps::input::extract_zip_file(zip_file_path, cache_path);
-
-    return cache_path;
+    return std::filesystem::absolute(root_path / path).string();
 }
-
 } // anonymous namespace
 
 namespace hgps::input {
 
-DataSource::DataSource(std::string source)
-    : source_(std::move(source)), file_hash_(std::nullopt), validate_checksum_(false) {}
+DataSource::DataSource(std::string source) : source_(std::move(source)) {}
 
-DataSource::DataSource(std::string source, const std::filesystem::path &root_path,
-                       std::optional<std::string> file_hash)
-    : source_(try_rebase_path(std::move(source), root_path)), file_hash_(std::move(file_hash)),
-      validate_checksum_(true) {}
+DataSource::DataSource(std::string source, const std::filesystem::path &root_path)
+    : source_(try_rebase_path(std::move(source), root_path)) {}
+
+std::string DataSource::compute_file_hash(const std::filesystem::path &zip_file_path) const {
+    return hgps::compute_sha256_for_file(zip_file_path);
+}
```

Comment:
Is this unvalidated hash used in caching? Else not sure why it's needed in unvalidated datasource.

Reply:
Yep, that's it.